### DISCUSSION TITLE
Heimdall v3 Capabilities

### DISCUSSION THREAD

**Discussion by georgedias** [mitre/heimdall2#5447] [Ideas]
https://jira.mitre.org/browse/SAFWK-69 (backend) and https://jira.mitre.org/browse/SAFWK-70 (frontend)

# Discussion board for capabilities to be added, modified for Heimdall v3 (Vue3)

### Database
- Database data content encryption
  - For the entire application (toggled)
  - For Groups
  - For Users
- Database management 
  - Record deletion, individually or bulk
  - Backup and Restore
  - Edit, delete entries (evaluations, profiles) 
  - Garbage collection (use VACUUM)
  - Auto generate [API](https://www.youtube.com/watch?v=sNh9PoM9sUE)
- PostgresSQL [Foreign Data Wrappers](https://www.postgresql.org/docs/9.3/postgres-fdw.html)
  - Use postgres_fdw to query a remote database

### Interfaces (external services, database, local file system, etc)
-  Add the option to filter (search) database entries by date range - when filtering by date or data range all other filter capabilities are not applicable

### Database
- Use Store Procedures (SP) for common queries

### Interfaces (external services, database, local file system, etc)
- Redesign pagination capability to use Vue 3 
  - [Server side paginate and sort](https://hc200ok.github.io/vue3-easy-data-table-doc/features/server-side-paginate-and-sort.html)
  - [Vuetify Server side tables](https://vuetifyjs.com/en/components/data-tables/basics/#server-side-tables)

**Comment by p-oneil** [comment]
Availability/scalability - support a load balancing configuration.
